platform-core-module/src/main/java/com/softicar/platform/core/module/program/execution/ProgramExecutionInserter.java

```diff
@@ -25,18 +27,24 @@ public AGProgramExecution insert() {
 
 	private AGProgramExecution insertAndUpdateCurrentExecution() {
 
-		program//
-			.setCurrentExecution(insertExecution())
-			.setQueuedAt(null)
-			.save();
-
-		return program.getCurrentExecution();
+		try (var transaction = new DbTransaction()) {
+			program.reloadForUpdate();
+			program//
+				.setCurrentExecution(insertExecution(program.getQueuedBy()))
+				.setQueuedBy(null)
+				.setQueuedAt(null)
+				.save();
+			var currentExecution = program.getCurrentExecution();
```

Comment:
`currentExecution` could be inlined... won't make a difference, and looks less weird:
```
transaction.commit();
return program.getCurrentExecution();
```